src/HealthGPS.Console/model_parser.cpp

```diff
@@ -140,13 +140,31 @@ std::unique_ptr<hgps::StaticLinearModelDefinition>
 load_staticlinear_risk_model_definition(const poco::json &opt, const host::Configuration &config) {
     MEASURE_FUNCTION();
 
-    // Risk factor linear models and correlation matrix.
-    std::vector<hgps::LinearModelParams> risk_factor_models;
+    // Correlation matrix.
     const auto correlations_file_info =
         host::get_file_info(opt["RiskFactorCorrelationFile"], config.root_path);
     const auto correlations_table = load_datatable_from_csv(correlations_file_info);
     Eigen::MatrixXd correlations{correlations_table.num_rows(), correlations_table.num_columns()};
 
+    // Weight quantiles.
+    const auto quantiles_female = load_datatable_from_csv(
+        host::get_file_info(opt["WeightQuantiles"]["female"], config.root_path));
+    const auto quantiles_male = load_datatable_from_csv(
+        host::get_file_info(opt["WeightQuantiles"]["male"], config.root_path));
+    std::map<hgps::core::Gender, std::vector<double>> weight_quantiles = {
+        {hgps::core::Gender::female, std::vector<double>(quantiles_female.num_rows())},
+        {hgps::core::Gender::male, std::vector<double>(quantiles_female.num_columns())}};
+    for (size_t j = 0; j < quantiles_female.num_rows(); j++) {
+        weight_quantiles[hgps::core::Gender::female][j] =
+            std::any_cast<double>(quantiles_female.column(0).value(j));
+    }
+    for (size_t j = 0; j < quantiles_male.num_rows(); j++) {
+        weight_quantiles[hgps::core::Gender::male][j] =
+            std::any_cast<double>(quantiles_male.column(0).value(j));
+    }
```

Comment:
I guess you intended the male vector to be `quantiles_male.num_rows()` in size?

That aside, it's better not to resize your vectors upfront, but to resize them as you go along. If you want to hint to the compiler that you know the final size you can use the `reserve()` method.

I think this should work:

```suggestion
    std::map<hgps::core::Gender, std::vector<double>> weight_quantiles = {
        {hgps::core::Gender::female, {}},
        {hgps::core::Gender::male, {}};
    weight_quantiles[hgps::core::Gender::female].reserve(quantiles_female.num_rows());
    weight_quantiles[hgps::core::Gender::male].reserve(quantiles_male.num_rows());
    for (size_t j = 0; j < quantiles_female.num_rows(); j++) {
        weight_quantiles[hgps::core::Gender::female].push_back(std::any_cast<double>(quantiles_female.column(0).value(j)));
    }
    for (size_t j = 0; j < quantiles_female.num_rows(); j++) {
        weight_quantiles[hgps::core::Gender::male].push_back(std::any_cast<double>(quantiles_male.column(0).value(j)));
    }
```

That said, this is rather ugly. I see this is where you wanted to use #253.

The problem is we get a `DataTypeColumn` here (the abstract base class), whereas your fix applies to `PrimitiveDataTypeColumn` -- I can't think of a better way to reorganise this API without digging into it a bit more though (which I'm happy to do, @jamesturner246!).

One option would be to use `dynamic_cast` to convert the `DataTypeColumn`s into `PrimitiveDataTypeColumn<double>`s, then we can use your fix in #253 to get at the underlying `std::vector`s. What do you reckon, @jamesturner246?

Reply:
PS -- The way you've done it is also fine (aside from the sizing problem) and if we're going to rework the API anyway, there's not much point in mucking about with it now if it works.

Reply:
Yep, this is where I was thinking on using that patch. 

I thought on using `push_back`, but for some reason I thought it was going to be less efficient that just allocating the space upfront, given that I knew it in advance, and just populating the elements.

Thanks for pointing the male/female issue. I'l fix it asap.

Reply:
If not using something similar to #253, the only other option I can think of is implementing my own parser to read those files, but it is a pity given that there is already perfectly fine functionality to do that.

Reply:
It's likely to be a really minor performance difference in this case, but the reason *not* to resize containers when creating them is because the contents then have to be default-initialised (in this case that just means they'll be zeroed). If you use `reserve()` instead, you still allocate memory but there's no initialisation required.

Reply:
> If not using something similar to #253, the only other option I can think of is implementing my own parser to read those files, but it is a pity given that there is already perfectly fine functionality to do that.

I *think* we could rework the API so that you could get the values out by doing something like:

```cpp
std::vector<double> my_values = column.values<double>();
```

If `column` contains `int`s or something else, you would get an exception.

Reply:
Keeping aside mentioning the specific `type`, this was my intention, but it seems I did it in the wrong place.

Reply:
You would need to update `PrimitiveDataColumnType` as you've done as well, but we also need to figure out how to change the base class to make use of your new `values()` method. Python makes this so much easier :laughing:

Reply:
I think this can wait until afterwards. Lots of changes going on there at the moment.